src/main/kotlin/net/lumalyte/lg/infrastructure/listeners/ProgressionEventListener.kt

```
@@ -314,4 +348,106 @@ class ProgressionEventListener : Listener, KoinComponent {
+
+    private fun flushPendingGuildXp() {
+        val snapshot = mutableListOf<Pair<GuildXpKey, Int>>()
+        pendingGuildXp.forEach { (key, counter) ->
+            val amount = counter.getAndSet(0)
+            if (amount > 0) snapshot.add(key to amount)
+        }
+
+        snapshot.forEach { (key, amount) ->
+            try {
+                progressionService.awardExperience(key.guildId, amount, key.source)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)